numba/cpython/numbers.py

```diff
@@ -361,6 +361,38 @@ def int_ne_impl(context, builder, sig, args):
     return impl_ret_untracked(context, builder, sig.return_type, res)
 
 
+def int_su_cmp(op):
+    def impl(context, builder, sig, args):
+        (left, right) = args
+        # This code is entirely too clever. It's taken from the NumPy source.
```

Comment:
Could you add a link to the NumPy source it's taken from?

Reply:
Reading through your description in the comments, I actually think this code is "just the right amount of clever" :slightly_smiling_face: